sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1375,6 +1375,12 @@ object SQLConf {
     .timeConf(TimeUnit.SECONDS)
     .createWithDefaultString(s"${5 * 60}")
 
+  val INTERRUPT_ON_CANCEL = buildConf("spark.sql.execution.interruptOnCancel")
+    .doc("When true, all running tasks will be interrupted if one cancels a query.")
+    .version("3.5.0")
```

Comment:
I think this should better be 4.0.0

Reply:
OK

Reply:
I'm fine to add this to 3.5.0 if it does not block some thing. Since 3.5.0 we change the default interruptOnCancel to true in thrift mode and we have new interruptOnCancel api at spark context and connect. I think it's a chance to also have same behavior at sql execution.

Reply:
Yeah, I sort of agree with that .. but it affects the whole Spark SQL that is much broader scope .. so I would like to target 4.0.0 so we can have some more time to take a look